engine/src/variable.cpp

```diff
@@ -119,7 +119,7 @@ bool MCVariable::encode(void *&r_buffer, uindex_t& r_size)
     }
     else if (MCExecTypeIsNumber(value . type))
     {
-        double t_value;
+        double t_value = 0;
```

Comment:
For the sake of clarity I suggest using 0.0 for double literals, and 0.0f for float literals.